publish.go

```diff
@@ -59,18 +59,21 @@ func RegisterPublisher(scheme string, factory PublisherFactory) {
 
 // PubMessage represents a single message for publishing.
 type PubMessage struct {
-	// ID is a message identifier.
+	// ID is an optional message identifier.
+	// It may not be supported by some implementations (then it is ignored).
+	// Or may be used just to calculate partition the message.
 	ID string `json:"id,omitempty"`
 
 	// Data is the message payload.
 	Data []byte `json:"data,omitempty"`
 
-	// Attributes contains key-value labels.
+	// Attributes contains optional key-value labels.
+	// It may not be supported by some implementations (then it is ignored).
 	Attributes map[string]string `json:"attributes,omitempty"`
 }
 
-// Topic is a publisher handle to a topic.
-type Topic interface {
+// PubTopic is a publisher handle to a topic.
+type PubTopic interface {
 	// Publish publishes a message to the topic.
 	Publish(context.Context, *PubMessage) error
 	// PublishBatch publishes a batch of messages to the topic.
```

Comment:
separate PR, but I think we should remove this for now, we can figure out something else later if we need it